tasks/scripting/src/main/java/org/infinispan/scripting/impl/ScriptingManagerImpl.java

```diff
@@ -148,11 +148,15 @@ boolean containsScript(String taskName) {
          AuthorizationManager authorizationManager = context.getCache().isPresent() ?
                SecurityActions.getAuthorizationManager(context.getCache().get().getAdvancedCache()) : null;
          if (authorizationManager != null) {
+            // when the cache is secured
             authorizationManager.checkPermission(AuthorizationPermission.EXEC, metadata.role().orElse(null));
          } else {
-            authorizer.checkPermission(AuthorizationPermission.EXEC, metadata.role().orElse(null));
+            if (context.getSubject().isPresent()) {
+               authorizer.checkPermission(context.getSubject().get(), AuthorizationPermission.EXEC);
```

Comment:
Katia, could you add a comment to explain why this is needed? 

Also, I noticed a similar pattern in `ServerTaskEngine.checkPermissions()`, maybe the same change is required there as well?